tools/integration/lib/compareDefinitions.js

```diff
@@ -0,0 +1,218 @@
+// (c) Copyright 2024, GitHub and ClearlyDefined contributors. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+function compareDocuments(staging, production, ignoredKeys, path = '') {
+  let differences = {}
+  let overallResults = []
+
+  staging = staging === undefined ? {} : staging
+  production = production === undefined ? {} : production
+
+  const keys = new Set([...Object.keys(staging), ...Object.keys(production)])
+
+  for (let key of keys) {
+    const currentPath = path ? `${path}.${key}` : key
+
+    if (shouldIgnore(currentPath, ignoredKeys)) {
+      continue
+    }
+
+    const stagingValue = staging[key]
+    const productionValue = production[key]
+
+    const comparison = compareValues(stagingValue, productionValue, ignoredKeys, currentPath)
+    overallResults.push(comparison.result)
+
+    if (comparison.result !== 'parity' && comparison.differences) {
+      for (let [resultType, diffs] of Object.entries(comparison.differences)) {
+        if (!differences[resultType]) {
+          differences[resultType] = []
+        }
+        differences[resultType] = differences[resultType].concat(diffs)
+      }
+    }
+  }
+
+  const overallResult = aggregateOverallResult(overallResults)
+
+  return {
+    overallResult,
+    differences
+  }
+}
+
+function compareValues(val1, val2, ignoredKeys, path) {
```

Comment:
Perhaps a change to expected and actual as the variable names?  Or devActual and prodActual since they are both coming from live data.  Although if used with static and dynamic, may devActual and expected, where expected comes from either fixtures for static tests or prod actual for dynamic tests.